cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -105,6 +108,8 @@ protected boolean removeEldestEntry(Map.Entry<String, AnomalyState> eldest) {
     _ongoingAnomalyDurationSumForAverageMs = 0;
     _numSelfHealingStarted = new AtomicLong(0L);
     _numSelfHealingFailedToStart = new AtomicLong(0L);
+    _numSelfHealingStartedPerAnomaly = new HashMap<>();
```

Comment:
Do you need to do anything in the `refreshMetrics` method to reset these maps? There seems to be a `numSelfHealingStarted()` call in there as part of creating a new `AnomalyMetrics` instance.

I don't know if that is relevant but you should check if these maps need to be reset.